lib/rubocop/cop/dependency/over_boundary.rb

```diff
@@ -18,6 +18,8 @@ module Dependency
       #
       class OverBoundary < Base
         def on_const(node)
+          return if node.parent&.const_type?
```

Comment:
When `X::Y::X` appears in the source code, this `on_const` is called three times.

Results of `node.parent&.const_type?` are as follows:

pattern|result
--|--
`X`| true
`X::Y`| true
`X::Y::X`| false

So when BannedConstPatterns is `X` and 21L is not written, three offences appear like this:

```sh
X::Y::Z
^^^^^^^

X::Y::Z
^^^^

X::Y::Z
^
```